bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -1300,6 +1301,41 @@ else if (isPlayingStream(currentURI) || isPlayingRadioStartedByAmazonEcho(curren
                 }
             }
 
+            else if (isPlayingRadioApp(currentURI)) {
+                if (currentUriMetaData != null) {
+                    title = currentUriMetaData.getTitle();
+                    resultString = title;
+                    if (currentTrack != null) {
+                        String[] contents = currentTrack.getStreamContent().split("\\|");
+                        String songTitle = null;
+                        for (int i = 0; i < contents.length; i++) {
+                            if (contents[i].startsWith("TITLE ")) {
+                                String val = contents[i].substring(6).trim();
+                                if (!val.startsWith("Advertisement_")) {
+                                    songTitle = val;
+                                }
+                            }
+                            if (contents[i].startsWith("ARTIST ")) {
+                                artist = contents[i].substring(7).trim();
+                            }
+                            if (contents[i].startsWith("ALBUM ")) {
+                                album = contents[i].substring(6).trim();
+                            }
+                        }
+                        if (artist != null && !artist.isEmpty()) {
+                            resultString += " - " + artist;
+                        }
+                        if (album != null && !album.isEmpty()) {
+                            resultString += " - " + album;
+                        }
+                        if (songTitle != null && !songTitle.isEmpty()) {
+                            resultString += " - " + songTitle;
+                        }
```

Comment:
Consider if this logic could be extracted to private method for keeping this method simpler.

Reply:
Not easily as it sets 4 variables, title, artist, album and resultString.

Reply:
In fact, I could create several methods. I will do that.

Reply:
Finally not because it will imply to duplicate the end of the method.

Reply:
Ok, I am going to create an object and it should be possible.

Reply:
Finally done